### DISCUSSION TITLE
[2024-02-12] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#9016] [Weekly Demo Day]
- James: Allow Filter_Condition to be inverted. #8861
- James: Add some ICON tags and a few new Icons. #8996
- James: Text Editor shows up for Text only parameters. #9014
- James: Update widget for various table functions. #9017 
- James: Applied ICONs from @Cassandra-Clark. #9023 
- James: Refactor aggregate to have group_by parameter. #9027 
- Radosław: Check type of `self` in static dispatch #8867
  - Initial implementation of `Self` type resolution - needed to handle some edge cases, finally all tests are passing.
  - Still waiting for a fix to engine benchmarks (#9026) which are currently broken to be able to analyze the performance impact.
- Radosław: Write support for S3 #8921
  - Addressing review, creating followup tasks
- Radosław: Helping @Cassandra-Clark with the license review tool for #8907
- Greg: Disable tests that fail because of locale-specific Postgres Unicode collation differences #8869
- Greg: Implement Table.replace for the in-memory backend #8935
- Greg: Implement Table.replace for the database backend #8986
- Greg: Re-fetch Warnings in Warning.getWarnings() to correctly wrap them #9011
- Cassandra: add google analytics authentication and reporting to enso #8907
- Cassandra: added new ICON tags to methods #8926
- Cassandra: added new ICON tags to methods and chose new icons #8916
- Cassandra: refined `table.running` design #8196 
- Ad: Add to_xml component #8979
- Ad: Merge empty_error #9000
- Ad: Remove _new and actually run the new tests #9006
- Jaroslav: Execute and debug `.enso` files with `bin/enso` in VSCode #8923
- Jaroslav: Store whole IR.Module in .bindings cache #8924
- Jaroslav: **Improving GraalVM** polyglot API to better _suite Enso needs_: https://github.com/oracle/graal/pull/8266
- Jaroslav: Better `runEngineDistribution --run test/Base_Tests "filter"` #8940
- Jaroslav: Keep only module-info.class in the root of runtime-fat-jar #8961
- Hubert: Missing expression updates on restart #8689 
- Hubert: Minor cleanups to docker image #8925
- Hubert: Further optimizations to test resources #8936
- Hubert: Disable flaky test #8967
- Pavel: Remove fansi dependency from runtime-compiler #8847
- Pavel: Fix teardown in SQLite File spec #8918
- Pavel: Remove akka from downloader and from runtime #8953
- Pavel: Refactor stdlib tests to the builder API #8968
- Pavel: Fix incremental compilation of runtime/test #8975
- Pavel: Merge Test_New into Test #8991
- Pavel: Remove obsolete Test_New import #9013
- Dmitry: Local project manager work against any folder #8985
- Dmitry: Fix NPE when recomputing the context #8999
- Dmitry: Cleanup execution contexts when the client disconnects #9004
- Dmitry: Escape default text arguments in suggestions database #9010

**Comment by AdRiley** [comment]
Table.to_xml component

https://github.com/enso-org/enso/assets/1720119/d0c61ee5-c301-4f04-a6c4-3b3e142df35c

**Comment by JaroslavTulach** [comment]
[Official results](https://enso-org.github.io/engine-benchmark-results/stdlib-benchs.html) of benchmarking integration of #8924 indicate speedup by more than 100ms!

![obrazek](https://github.com/enso-org/enso/assets/26887752/7f2dd572-0ab3-4ceb-897c-e1be8d9543fe)

- E.g. the speedup isn't just because of #8698

**Comment by JaroslavTulach** [comment]
@ola-lis updated [Enso VSCode extension](https://marketplace.visualstudio.com/items?itemName=Enso.enso4vscode) to version 1.33.29 that provides support for **debugging a single Enso file**:

![Select enso executable](https://github.com/enso-org/enso/assets/26887752/4e1d0666-634d-4fb8-bf61-6dbf765311e8)

Read more in the [documentation](https://github.com/enso-org/enso/blob/d87ba6e3cf7804e72909db1a984e1b53c4b0c004/tools/enso4igv/README.md#debugging-a-single-enso-file)!
